### ISSUE TITLE
Not fails if Check returns an error

### ISSUE THREAD

**Issue by sgargan** [go-check/check#10]
Not sure if this is the expected behavior, but if a checker returns an error message this will cause a surrounding 'Not' to fail.

``` go
type failsWithError struct {
  *CheckerInfo
}

var FailsWithError Checker = &containsChecker{
  &CheckerInfo{Name: "FailsWithError", Params: []string{"obtained", "expected"}},
}

func (checker *failsWithError) Check(params []interface{}, names []string) (result bool, error string) {
  return false, "Error message explaining failing check"
}
```

running this checker with not will fail

``` go
c.Assert("somestring", Not(FailsWithError), "fails")
```

Should the Not fail in this way? Not returning the message loses information about the reason for the failure. 

Modifying the Not check to clear the error if the result is true lets the test pass

``` go
func (checker *notAndIgnoreError) Check(params []interface{}, names []string) (result bool, error string) {
    result, error = checker.sub.Check(params, names)
  result = !result
    if result {
    error = ""
  }
    return
}
```

 is this how the Not should behave?

**Comment by niemeyer** [comment]
The behavior is correct. The error means the use of the checker was inappropriate and the test must be fixed, rather than the test has failed.
